roberta.py

```diff
@@ -0,0 +1,232 @@
+        batch_inputs, batch_masks, batch_labels = \
+                        tuple(b for b in val_batch)
+        outputs = self(batch_inputs, batch_masks)    
+        val_loss = self.loss(outputs.squeeze(1).float(), 
+                        batch_labels.float())
+        self.log("val/loss", val_loss)
+        return val_loss
+
+class RobertaRegressor(nn.Module):
+    def __init__(self, len_tokenizer, dropout=0.2, model_name = 'roberta-base'):
+        super(RobertaRegressor, self).__init__()
+        D_in, D_out = 768, 1
+        self.roberta = RobertaModel.from_pretrained(model_name, num_labels=1)
+        self.roberta.resize_token_embeddings(len(len_tokenizer))
+        self.regressor = nn.Sequential(
+            nn.Dropout(dropout),
+            nn.Linear(D_in, D_out))
+        
+    def forward(self, input_ids, attention_masks):
+        outputs = self.roberta(input_ids, attention_masks)
+        class_label_output = outputs[1] #Last layer hidden-state of the first token of the sequence (classification token) (but also can be used for regression)
+        outputs = self.regressor(class_label_output)
+        return torch.tanh(outputs) 
+    
+if __name__=="__main__":
+    #TODO move hyperparamethers to a seperate CONFIG file
+    test_size = 0.2
+    val_size = 0.5
+    batch_size = 8
+    epochs = 10
+
+    model_name = 'roberta-base'
+
+    text_cleaner = DataPreprocessor()
+    df_processor = DFProcessor(filename='ruddit_with_text.csv')
+
+    new_df = df_processor.process_df(text_cleaner)
+
+    data_augmentator = TextAugmenterForBert(tokenizer_name=model_name, df = new_df)
+    #here we do everything tokenizer-related i.e. encoding corpus, getting input ids etc.
+    input_ids, labels, attention_mask = data_augmentator.augment_data()
+
+    #80% train 10% test 10% val is proposed
+    train_inputs, test_inputs, train_labels, test_labels, train_masks, test_masks = \
+            train_test_split(input_ids, labels, attention_mask, test_size=test_size, shuffle=True)
+    val_inputs, test_inputs, val_labels, test_labels, val_masks, test_masks = \
+            train_test_split(test_inputs, test_labels, test_masks, test_size=val_size)
+    
+    train_dataloader = create_dataloaders(train_inputs, train_masks, 
+                                        train_labels, batch_size)
+    test_dataloader = create_dataloaders(test_inputs, test_masks, 
+                                        test_labels, batch_size)
+    val_dataloader = create_dataloaders(val_inputs, val_masks, 
+                                        val_labels, batch_size)
+    
+    #here declare bert-like model, just change the "RobertaModel" into something else.
+    bertlike_model = RobertaModel.from_pretrained(model_name, num_labels=1)
+    bertlike_model.resize_token_embeddings(len(data_augmentator.tokenizer))
+    
+    model = Regressor(bertlike_model=bertlike_model)
+    wandb_logger = WandbLogger(project = "rudeness_determinator")
```

Comment:
Don't forget _wandb.login()_ :)

Reply:
Can't we use wandb login in console?

Reply:
We can, but it's safer to ensure that every file that uses wandb can log in, so you don't have to do it every time you execute a file. Also, the _wandb.login()_ call is idempotent, so it has no effect if you are already logged in.

Reply:
Alright. Updated it.